lib/fabrik/DirectorManager.js

```diff
@@ -211,6 +211,31 @@ class DirectorManager extends BaseManager {
                 previous_manifest: manifest
               }, opts.context);
               return;
+            })
```

Comment:
Refactor lines 200-238 to a new method. The logic in-between these lines breaks the code flows and readability. Methods should generally be small and concise and if there too much specifics going in, then such things must be generally moved into a seperate method of its own.

Reply:
Done!